npm/private/npm_translate_lock_generate.bzl

```diff
@@ -219,7 +222,7 @@ def npm_link_targets(name = "node_modules", package = None):
     npm_link_all_packages_bzl = [
         """\
 # buildifier: disable=function-docstring
-def npm_link_all_packages(name = "node_modules", imported_links = []):
+def npm_link_all_packages(name = "node_modules", imported_links = [], prod = False, dev = False):
```

Comment:
Any other ideas for this API?

It might be nice if we didn't have this scenario where 1/4 of the states is basically invalid and has an odd non-intuitive default. Maybe just throw when `false/false` and then the other 3 states are valid and understandable?

`dev = True|False|None` or something like that is one alternative, although the word "dev" in the API without "prod" in the API is confusing imo. Or `type = "dev"|"prod"|None` but I'd want a term better then "type".

Reply:
Actually is this API simply a copy of `npm_translate_lock`? In which case we should do exactly as you have and my suggestion can be considered in rules_js v3 or something like that...

Reply:
Yeah, it's just following the `prod`/`dev` API from `npm_translate_lock`. I agree that an API that excludes invalid states by design would be better for v3.

Reply:
This current API does not allow a single BUILD to have a mix of these though, correct? If you wanted to include dev deps for some targets (such as tests) but not for others (such as libraries).

Would something such as `:node_modules[__{dev,prod}]` as the API also solve this?

Reply:
It does allow a mix. You could write `npm_link_targets()` and `npm_link_targets(prod = True)` and npm_link_targets(dev = True)` in the same BUILD if you wanted.

Reply:
I was thinking you can't have multiple `npm_link_all_packages()` targets though...?

Reply:
Do we need the `npm_link_all_packages` API changes in this PR at all?

Reply:
I can't remember why I included the `npm_link_all_packages` changes originally; I guess just for parallelism with `npm_link_targets`. We're only using `npm_link_targets(prod = True)`.